src/daft-distributed/src/pipeline_node/translate.rs

```diff
@@ -28,6 +28,71 @@ use crate::{
     stage::StageConfig,
 };
 
+/// Check if a logical plan can be translated to a map-only pipeline
+/// This function examines the logical plan structure to determine if it contains
+/// any operations that require data movement or blocking behavior
+fn is_map_only_pipeline(logical_plan: &LogicalPlanRef) -> bool {
```

Comment:
@colin-ho @srilman Unsure if I got this one right. My idea is that we can inspect the optimized logical plan when we create the `ScanTask` node. Our inspection can walk the logical plan from the scan node and, if it encounters anything that's not map-only, indicate as such. We can pass this value along into the ScanTask node by adding a boolean flag into the `struct` and checking that at runtime. Thoughts?